catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -517,6 +520,164 @@ public void init() {
             return "";
         });
 
+        /*
+         * Notes may have a virtual parent, which means the parent is not directly accessible to
+         * the framework. For example, the parent may be stored as a text field inside another
+         * metacard. For non-virtual parents, the parent must be retrievable by the framework.
+         * If the call does not set the isVirtual flag, then it defaults to 'false'.
+         */
+        post("/notes", (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
+            String parentID = incoming.get("parent")
+                    .toString();
+            String comment = incoming.get("note")
+                    .toString();
+            Boolean isVirtual = Boolean.parseBoolean(incoming.getOrDefault("isVirtual", "false")
+                    .toString());
+            String user = getSubjectEmail();
+            if (user == null) {
+                return util.getResponseWrapper("error",
+                        "You are not authorised to create notes! A user email is required. "
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            //Test if parent metacard ID is valid and comment is not empty
```

Comment:
In this case, valid means that the metacard exists. BTW: It's not actually used by my code because I set the isVirtual flag. But I'm expecting the PTE code on metacard-notes to get merged into ddf and I wanted to minimize the eventual merge conflicts.